### DISCUSSION TITLE
Composite Actions in private GitHub Organization

### DISCUSSION THREAD

**Discussion by johan-lindqvist** [renovatebot/renovate#19313] [Request Help] [answered]
We're using Renovate for our internal organization at work for our internal packages and it works great. We also have some internal composite GitHub actions that we'd like to use Renovate for but I'm having trouble getting the authorization setup correct. It finds the public GitHub actions fine and updates them, but it's getting errors saying that it can't find the internal GitHub actions:

```
DEBUG: Datasource unknown error
{
  "datasource": "github-tags",
  "packageName": "my-org/repository",
  "err": {
    "message": "Could not resolve to a Repository with the name 'my-org/repository'.",
    "stack": "Error: Could not resolve to a Repository with the name 'my-org/repository'.\n    at GithubGraphqlDatasourceHelper.doRawQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-helper.js:89:29)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async GithubGraphqlDatasourceHelper.doShrinkableQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-helper.js:138:26)\n    at async GithubGraphqlDatasourceHelper.doPaginatedQuery (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-helper.js:158:33)\n    at async GithubGraphqlDatasourceHelper.getItems (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-helper.js:188:21)\n    at async Function.query (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/datasource-helper.js:48:23)\n    at async queryTags (/home/ubuntu/renovateapp/node_modules/renovate/dist/util/github/graphql/index.js:8:17)\n    at async GithubTagsDatasource.getReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/github-tags/index.js:65:22)\n    at async getRegistryReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:57:17)\n    at async fetchReleases (/home/ubuntu/renovateapp/node_modules/renovate/dist/modules/datasource/index.js:240:23)"
  }
}
```

I'm assuming since I haven't setup the credentials to be used for the `github-tabs` datasource.

For our internal packages we have a `hostRules` config looking like this:
```json
{
  "hostRules": [
    {
      "matchHost": "nuget.pkg.github.com",
      "username": "<redacted>",
      "encrypted": {
        "password": "<redacted>"
      }
    }
  ]
}
```

But for internal composite GitHub actions I'm a bit lost as how I could setup the config to only use the credentials when fetching the internal composite GitHub actions, since I don't want it to use the credentials when fetching the public GitHub actions. And I couldn't find a way to combine package rules and host rules to solve the issue.

Any ideas how I can change my configuration to use `hostRules` for internal composite GitHub actions but not for public ones?

**Comment by rarkins** [comment, marked as answer]
How are you running Renovate, e.g. self-hosted?

**Reply by johan-lindqvist** [reply]
Using the Renovate GitHub App, so not self-hosted

**Reply by rarkins** [reply]
is 'my-org' the same org as the repo, or in a different org? And if in same org, do you have Renovate installed into it? If same org, both private, and both installed then it should "just work".

**Reply by johan-lindqvist** [reply]
Everything in same org, the organization requires authentication (using GitHub Enterprise Cloud, not the self-hosted GHES) so it doesn't work for NuGet packages either if I don't have the `hostRules`. So I assume the problem is that renovate doesn't know that it needs to use the credentials for the github actions since the `hostRules` are only specified to use the credentials if the host matches `nuget.pkg.github.com`

**Reply by johan-lindqvist** [reply]
I noticed that both repos did not have renovate enabled, testing now if that solves the issue

**Reply by johan-lindqvist** [reply]
Seems to be like you said, when renovate is enabled on both repos it "just works". Thanks for the quick help!
